swift/Lottie.swift

```diff
@@ -127,27 +128,61 @@ public class Lottie {
     }
 
     /// Prepares the canvas for rendering by setting the frame, adjusting the picture size, and optionally applying a crop.
-    private func prepareCanvasForRendering(frameIndex index: Int, size: CGSize, crop: CGRect?) {
+    private func prepareCanvasForRendering(frameIndex index: Int, size: CGSize, crop: CGRect?) throws {
         tvg_animation_set_frame(animation, Float(index))
 
         let picture = tvg_animation_get_picture(animation)
         tvg_picture_set_size(picture, Float(size.width), Float(size.height))
 
         if let crop {
-            let cropShape = tvg_shape_new()
+            try applyCropping(crop, relativeTo: size)
+        }
+
+        tvg_canvas_update_paint(canvas, picture)
+    }
+
+    /// Applies cropping to the Lottie animation by resizing and translating the picture to fit within a specified cropping rectangle, relative to a given size.
+    /// Note: The function ensures that the cropped area is scaled and positioned correctly to fit within the specified size, maintaining the aspect ratio of the cropped area.
+    private func applyCropping(_ crop: CGRect, relativeTo size: CGSize) throws {
```

Comment:
i guess `tvg_animation_get_picture(animation)` always returns the same picture pointer, otherwise this may not work. 

so maybe we pass the picture pointer as the method parameter is better to avoid confusion?